### PR TITLE
Fix: regression in identityoperator()+sparse_matrix due to the move to Eye #109

### PR DESCRIPTION
**Pull request by SatyaBade12**

### CODE REVIEW

File: test/test_operators_sparse.jl [hunk 1]
```diff
@@ -117,6 +117,21 @@ IEye = identityoperator(b_l)
 @test sparse(IEye) == I
 Icomp = identityoperator(b1a) ⊗ identityoperator(b2a) ⊗ identityoperator(b3a)
 @test IEye == Icomp
+for IEye in (IEye, -IEye, IEye, dagger(IEye))
```
Comment (amilsted) [inline test/test_operators_sparse.jl:120]: IEye is tested twice here?

**Comment by amilsted** [comment]
It seems I probably missed this because `SquareEye` behaves differently to `Eye` under these operations. The former behaves correctly in sums and krons with sparse matrices, the latter does not.

`identityoperator(some_basis)` ought to be producing a SquareEye, but apparently it isn't. In any case, the general Eye case definitely needed fixing!

File: test/test_operators_sparse.jl (hunk 1 above)
Comment (Krastanov) [reply test/test_operators_sparse.jl:120]: yup, it was supposed to be an adjoint, will fix

**Comment by Krastanov** [comment]
@amilsted , I added tests for non-square operators. If this is green on CI I will merge it and we can then follow up with #111 (which would now have some broken tests)

**Pull request merged**